java-frontend/src/main/java/org/sonar/java/model/JParser.java

```diff
@@ -346,14 +346,14 @@ private void declaration(@Nullable IBinding binding, Tree node) {
     if (binding == null) {
       return;
     }
-    sema.declarations.put(binding, node);
+    sema.declarations.put(sema.declarationBinding(binding), node);
```

Comment:
IMO for declarations `binding == sema.declarationBinding(binding)` already.

Reply:
MMh indeed. For declaration maybe it does not make sense.